frontend/src/features/workspace/components/AdminFeedbackModal/AdminFeedbackModalContent.tsx

```diff
@@ -0,0 +1,175 @@
+    useAdminFeedbackMutation()
+
+  const handleRatingClick = useCallback(
+    (rating: number) => {
+      createAdminFeedbackMutation
+        .mutateAsync(rating)
+        .then((data) => setFeedbackId(data._id))
+      setContentState(FeedbackModalContentState.CallForComment)
+    },
+    [createAdminFeedbackMutation, setFeedbackId, setContentState],
+  )
+
+  const handleCommentClick = useCallback(
+    (data: AdminFeedbackCommentForm) => {
+      if (feedbackId && !!data.comment) {
+        updateAdminFeedbackMutation.mutateAsync({
+          feedbackId,
+          comment: data.comment,
+        })
+      }
+      onClose()
+    },
+    [feedbackId, updateAdminFeedbackMutation, onClose],
+  )
+
+  const handleCallForCommentClick = () =>
+    setContentState(FeedbackModalContentState.CommentBox)
+
+  return (
+    <ModalBody px="1.5rem" py="1rem">
+      <AdminFeedbackModalContentBuilder
+        state={contentState}
+        onRatingClick={handleRatingClick}
+        onCallForCommentClick={handleCallForCommentClick}
+        onCommentClick={handleCommentClick}
+      />
+    </ModalBody>
+  )
+}
+
+const AdminFeedbackRatingContent = ({
+  onRatingClick,
+}: {
+  onRatingClick: (rating: number) => void
+}) => {
+  return (
+    <Stack direction="row" alignItems="center" gap="2.2rem">
+      <Text textStyle="h6">How was your form building experience?</Text>
+      <IconButton
+        variant="clear"
+        icon={<GoThumbsup />}
+        colorScheme="theme-blue"
+        aria-label="Good"
+        onClick={() => onRatingClick(AdminFeedbackRatingUp)}
+      />
+      <IconButton
+        variant="clear"
+        icon={<GoThumbsdown />}
+        colorScheme="theme-red"
+        aria-label="Bad"
+        onClick={() => onRatingClick(AdminFeedbackRatingDown)}
+      />
+    </Stack>
+  )
+}
+
+const AdminFeedbackCallForCommentContent = ({
+  onLinkClick,
+}: {
+  onLinkClick: () => void
+}) => {
+  return (
+    <Text textStyle="h6">
+      Thank you, you're the best!{' '}
+      <Link onClick={onLinkClick}>Want to tell us more?</Link>
+    </Text>
+  )
+}
+
+const AdminFeedbackCommentContent = ({
+  onCommentClick,
+}: {
+  onCommentClick: (data: AdminFeedbackCommentForm) => void
+}) => {
+  const { handleSubmit, register } = useForm<AdminFeedbackCommentForm>()
+  const isMobile = useIsMobile()
+  return (
+    <Stack w={isMobile ? undefined : '28.5rem'}>
+      <Flex alignItems="flex-start">
+        <ModalHeader px="0rem">Great!</ModalHeader>
+        <ModalCloseButton />
+      </Flex>
+      <Text textStyle="body-2">
+        Tell us about your form building experience in more detail!
+      </Text>
+      <Textarea
+        placeholder="Form is awesome"
+        mt="1rem"
+        {...register('comment')}
+      />
+      <Flex alignItems="flex-end" flexDirection="column">
+        <Button mt="1rem" float="right" onClick={handleSubmit(onCommentClick)}>
+          Submit
+        </Button>
+      </Flex>
```

Comment:
Eh? For this, submission should be blocked if the text area is empty right? Based on [figma comment](https://www.figma.com/file/CMq0CJtawn7NGM7Zdpi8Ti?node-id=17430:181522#472371976)

Reply:
Okay actually, let me shift it to a separate issue, 

Alongside the placeholder of the `Textarea` in the spirit of pushing this out fast. As it doesn't exactly break the flow

Reply:
Yea okays sure 👍